File: backend/src/main/java/com/smvita/computerseekho/entity/BatchAlbumImage.java
package com.smvita.computerseekho.entity;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.time.LocalDate;

@Entity
@Table(name = "batch_album_images")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class BatchAlbumImage {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "image_id")
    private Integer imageId;

    @ManyToOne(fetch = FetchType.LAZY, optional = false)
    @JoinColumn(name = "album_id", nullable = false)
    private BatchAlbum album;

    @Column(name = "image_url", nullable = false, length = 500)
    private String imageUrl;

    @Column(name = "caption", length = 255)
    private String caption;

    /** Nullable FK — ON DELETE SET NULL, so removing a staff member doesn't take photos with them. */
    @Column(name = "uploaded_by")
    private Integer uploadedBy;

    @Column(name = "upload_date", nullable = false)
    private LocalDate uploadDate = LocalDate.now();

    @Column(name = "display_order", nullable = false)
    private Integer displayOrder = 0;

    @Column(name = "is_active", nullable = false)
    private Boolean isActive = true;
}

File: dotnet_backend/backend/Models/BatchAlbumImage.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace ComputerSeekho.Models;

/// <summary>A photo inside a batch album.</summary>
[Table("batch_album_images")]
public class BatchAlbumImage
{
    [Key]
    [Column("image_id")]
    public int ImageId { get; set; }

    [Column("album_id")]
    public int AlbumId { get; set; }

    [Column("image_url")]
    [Required, MaxLength(500)]
    public string ImageUrl { get; set; } = string.Empty;

    [Column("caption")]
    [MaxLength(255)]
    public string? Caption { get; set; }

    [Column("uploaded_by")]
    public int? UploadedBy { get; set; }

    [Column("upload_date")]
    public DateOnly UploadDate { get; set; } = DateOnly.FromDateTime(DateTime.Today);

    [Column("display_order")]
    public int DisplayOrder { get; set; }

    [Column("is_active")]
    public bool IsActive { get; set; } = true;

    [ForeignKey(nameof(AlbumId))]
    public BatchAlbum? Album { get; set; }
}
